src/server/server_family.cc

```diff
@@ -491,20 +491,19 @@ void ServerFamily::Init(util::AcceptServer* acceptor, std::vector<facade::Listen
   if (ReplicaOfFlag flag = GetFlag(FLAGS_replicaof); flag.has_value()) {
     service_.proactor_pool().GetNextProactor()->Await(
         [this, &flag]() { this->Replicate(flag.host, flag.port); });
-    return;  // DONT load any snapshots
-  }
-
-  const auto load_path_result = snapshot_storage_->LoadPath(flag_dir, GetFlag(FLAGS_dbfilename));
-  if (load_path_result) {
-    const std::string load_path = *load_path_result;
-    if (!load_path.empty()) {
-      load_result_ = Load(load_path);
-    }
-  } else {
-    if (std::error_code(load_path_result.error()) == std::errc::no_such_file_or_directory) {
-      LOG(WARNING) << "Load snapshot: No snapshot found";
+  } else {  // load from snapshot only if --replicaof is empty
+    const auto load_path_result = snapshot_storage_->LoadPath(flag_dir, GetFlag(FLAGS_dbfilename));
+    if (load_path_result) {
+      const std::string load_path = *load_path_result;
+      if (!load_path.empty()) {
+        load_result_ = Load(load_path);
+      }
     } else {
-      LOG(ERROR) << "Failed to load snapshot: " << load_path_result.error().Format();
+      if (std::error_code(load_path_result.error()) == std::errc::no_such_file_or_directory) {
+        LOG(WARNING) << "Load snapshot: No snapshot found";
+      } else {
+        LOG(ERROR) << "Failed to load snapshot: " << load_path_result.error().Format();
+      }
     }
```

Comment:
Can you wrap this in a function, so that `Init()` is simpler?

Reply:
done